server/views/pages/goal/create/index.njk

```diff
@@ -23,37 +23,84 @@
-            <p class="govuk-hint">Goals should be SMART, meaning 'specific', 'measurable', 'achievable', 'relevant' and 'time-bound'. SMART goals allow progress to be better measured and evaluated.</p>
-            <p class="govuk-hint">You can add the steps to achieve the goal on the next page.</p>
-          {% endset %}
-
-          {{ govukTextarea({
-            name: "title",
-            id: "title",
-            value: form.title,
-            label: {
-              text: titleLabelHtml,
-              classes: "govuk-label--l",
-              isPageHeading: true
+        <input type="hidden" name="_csrf" value="{{ csrfToken }}" />
+        <input type="hidden" name="prisonNumber" value="{{ form.prisonNumber }}" />
+
+        {% set titleLabelHtml %}
+          Describe the goal you want to create for {{ prisonerSummary.firstName }} {{ prisonerSummary.lastName }}
+        {% endset %}
+
+        {% set hintTextHtml %}
+          <p class="govuk-hint">Goals should be SMART, meaning 'specific', 'measurable', 'achievable', 'relevant' and 'time-bound'. SMART goals allow progress to be better measured and evaluated.</p>
+          <p class="govuk-hint">You can add the steps to achieve the goal on the next page.</p>
+        {% endset %}
+
+        {{ govukTextarea({
+          name: "title",
+          id: "title",
+          value: form.title,
+          label: {
+            text: titleLabelHtml,
+            classes: "govuk-label--l",
+            isPageHeading: true
+          },
+          hint: {
+            html: hintTextHtml
+          },
+          errorMessage: errors | findError('title')
+        }) }}
+
+        {% set targetCompletionDateItems = [] %}
+
+        {% for futureGoalTargetDate in futureGoalTargetDates %}
+          {% set targetCompletionDateItems = (targetCompletionDateItems.push({
+            "value": futureGoalTargetDate.value,
+            "text": futureGoalTargetDate.text
+            }), targetCompletionDateItems)
+          %}
+        {% endfor %}
+
+        {% set anotherDateHtml %}
+          {{ govukDateInput({
+            id: "another-date",
+            namePrefix: "another-date",
+            fieldset: {
+              legend: {
+                text: "What date are they aiming to achieve this by?",
+                classes: "govuk-visually-hidden"
+              }
             },
             hint: {
-              html: hintTextHtml
-            },
-            errorMessage: errors | findError('title')
+              text: "For example, 27 3 2007"
+            }
           }) }}
+        {% endset -%}
 
-          {{ govukButton({
-            text: "Continue to add steps"
-          }) }}
+        {% set targetCompletionDateItems = (targetCompletionDateItems.push({
+          "value": "another-date",
+          "text": "set another date",
+          "conditional": {
+            "html": anotherDateHtml
+          }
+          }), targetCompletionDateItems)
+        %}
```

Comment:
Ohh, this is messier than I'd anticipated, but I can see what it's doing and why ...

I wonder if we can do away with the `targetCompletionDateItems` array altogether, and simply push to the `futureGoalTargetDates` array?

Would something like this work?
```
{{ futureGoalTargetDates.push({
          "value": "another-date",
          "text": "set another date",
          "conditional": {
            "html": anotherDateHtml
          }
          })
}}
```